test/integration/dialects/mysql/connector-manager.test.js

```diff
@@ -87,8 +87,11 @@ if (dialect === 'mysql') {
         .then(connection => {
           // Save current connection
           conn = connection;
-          // simulate a unexpected end
-          connection.close();
+          // simulate a unexpected end from MySQL2
+          connection._protocolError = new Error('Connection lost: The server closed the connection.');
+          connection._protocolError.fatal = true;
+          connection._protocolError.code = 'PROTOCOL_CONNECTION_LOST';
+          connection._notifyError(connection._protocolError);
```

Comment:
What I don't like here is that we are using private underscore-prefixed methods that could therefor break our builds in a semver minor/patch.